File: examples/garden-k8s/clients/ios/BetterAuthBasicExample/Implementation/Time/Rfc3339.swift
import Foundation
import BetterAuth

class Rfc3339: ITimestamper {
    func format(_ when: Date) -> String {
        let formatter = ISO8601DateFormatter()
        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        return formatter.string(from: when)
    }

    func parse(_ when: Any) throws -> Date {
        if let date = when as? Date {
            return date
        }
        guard let string = when as? String else {
            throw ExampleError.invalidData
        }
        let formatter = ISO8601DateFormatter()
        formatter.formatOptions = [.withInternetDateTime, .withFractionalSeconds]
        guard let date = formatter.date(from: string) else {
            throw ExampleError.invalidData
        }
        return date
    }

    func now() -> Date {
        Date()
    }
}

File: examples/garden-k8s/services/app-rb/lib/app_encoding/rfc3339.rb
require 'time'

module AppEncoding
  class Rfc3339
    def format(when_time)
      when_time.utc.iso8601(3)
    end

    def parse(when_string)
      Time.iso8601(when_string)
    end

    def now
      Time.now.utc
    end
  end
end
